UniversalAlgebra/Terms/Setoid.lagda

```diff
@@ -74,48 +72,55 @@ module Environment (M : SetoidAlgebra α ℓ) where
 
  open SetoidAlgebra M
 
- open IsEquivalence renaming ( refl  to  reflE
-                             ; sym   to  symE
-                             ; trans to  transE )
-
  open Setoid        renaming ( refl  to  reflS
                              ; sym   to  symS
                              ; trans to  transS)
 
  -- Equality in M's interpretation of its sort.
- _≃_ : Den .Carrier → Den .Carrier → Type ℓ
- _≃_ = Den ._≈_
+ _≃_ : Domain .Carrier → Domain .Carrier → Type ℓ
+ _≃_ = Domain ._≈_
 
 
  -- An environment for Γ maps each variable `x : Γ` to an element of M.
  -- Equality of environments is defined pointwise.
  Env : Type χ → Setoid _ _
- Env Γ .Carrier                     = (x : Γ) → Den .Carrier
- Env Γ ._≈_ ρ ρ'                    = (x : Γ) → ρ x ≃ ρ' x
- Env Γ .isEquivalence .reflE      x = reflS  Den
- Env Γ .isEquivalence .symE     h x = symS   Den (h x)
- Env Γ .isEquivalence .transE g h x = transS Den (g x) (h x)
+ Env Γ = record { Carrier = (x : Γ) → Carrier Domain
+
+                ; _≈_ = λ ρ ρ' → (x : Γ) → ρ x ≃ ρ' x
+
+                ; isEquivalence =
+                   record { refl = λ _ → reflS Domain
+                          ; sym = λ h x → symS Domain (h x)
+                          ; trans = λ g h x → transS Domain (g x) (h x)
+                          }
+                }
```

Comment:
```
   where open IsEquivalence Domain
```
would add 1 line but simplify the code.

Reply:
Hmmm... I'm not sure what you mean here.  Adding that on line 96 after the } doesn't seem to work.